api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/TxMlQueries.java

```diff
@@ -25,106 +25,90 @@ public static String getPatientsWhoMissedAppointment(
-            + "	) pp"
-            + "	INNER JOIN ("
-            + "		SELECT p.patient_id, "
-            + "			("
-            + "				SELECT "
-            + "					e.encounter_id "
-            + "				FROM encounter e"
-            + "				INNER JOIN obs o ON e.encounter_id = o.encounter_id"
-            + "				WHERE "
-            + "					e.patient_id = p.patient_id"
-            + "					AND e.voided = 0"
-            + "					AND o.voided = 0"
-            + "					AND e.encounter_datetime <= :endDate"
-            + "					AND e.encounter_type IN (${adultoSequimentoEncounterType},${pediatriaSeguimentoEncounterType})"
-            + "				ORDER BY e.encounter_datetime DESC"
-            + "				LIMIT 1"
-            + "			) last_cl_encounter,"
-            + "			("
-            + "				SELECT "
-            + "					e.encounter_id "
-            + "				FROM encounter e"
-            + "				INNER JOIN obs o ON e.encounter_id = o.encounter_id"
-            + "				WHERE "
-            + "					e.patient_id = p.patient_id"
-            + "					AND e.voided = 0"
-            + "					AND o.voided = 0"
-            + "					AND e.encounter_datetime <= :endDate"
-            + "					AND e.encounter_type = ${pharmacyEncounterType}"
-            + "				ORDER BY e.encounter_datetime DESC"
-            + "				LIMIT 1"
-            + "			) AS latest_pharm_encounter_id,"
-            + "			("
-            + "					SELECT "
-            + "						e.encounter_id "
-            + "					FROM encounter e"
-            + "					INNER JOIN obs o ON e.encounter_id = o.encounter_id"
-            + "					WHERE "
-            + "						e.patient_id = p.patient_id"
-            + "						AND e.voided = 0"
-            + "						AND o.voided = 0"
-            + "						AND o.value_datetime <= :endDate"
-            + "						AND e.encounter_type = ${masterCardDrugPickupEncounterType}"
-            + "						AND o.concept_id = ${artPickupDateMasterCardConcept}"
-            + "					ORDER BY o.value_datetime DESC"
-            + "					LIMIT 1"
-            + "			) AS latest_dp_encounter_id				"
-            + "		FROM patient p"
-            + "	) last_encounters ON last_encounters.patient_id = pp.patient_id"
-            + "	WHERE pp.encounter_id IN (last_cl_encounter, latest_pharm_encounter_id, latest_dp_encounter_id)"
-            + "	GROUP BY pp.patient_id"
-            + ")all_patients "
-            + " WHERE "
-            + "	DATE_ADD(return_date, INTERVAL 28 DAY)  >= DATE_ADD(:startDate, INTERVAL -1 DAY) "
-            + "	AND DATE_ADD(return_date, INTERVAL 28 DAY) < :endDate";
+        " SELECT final.patient_id    "
+            + "             from(    "
+            + "                 SELECT    "
+            + "                     most_recent.patient_id, Max(most_recent.value_datetime) final_encounter_date    "
+            + "                 FROM   (SELECT fila.patient_id, o.value_datetime from (   "
+            + "                             SELECT enc.patient_id,    "
```

Comment:
@cfaife this query is not behaving the way you think it should, you doing a select with one table and group by with other table, you need to use the same thing!

> Select pa.patient_id instead of enc.patient_id

Reply:
@cfaife Was having problem with location id